### ISSUE TITLE
Error Handling Example

### ISSUE THREAD

**Issue by screwyprof** [deliveryhero/pipeline#3]
**Is your feature request related to a problem? Please describe.**
I'm trying to use this wonderful library in one of my project, but I'm not sure how to handle processor errors correctly.
At the moment if an errors occurs the processor.Cancel method is called, but how is it intended by the library to actually 
handle these errors by the caller?

**Describe the solution you'd like**
There should be a simple way to return an error channel from the processor.

**Describe alternatives you've considered**
Alternatively we have to pass an error channel to all the processor we use in the pipeline and send an error to that channel in the process.Cancel method.

**Additional context**
At the moment I've created a special Result type to be used by all the processors. It has an Err and a Value fields.
But the code is very verbose:

```go
type Result struct {
    Err error
    Value interface{}
}

func (p *SomeProcessor) Process(ctx context.Context, ins interface{}) (interface{}, error) {
	insSlice, err := toSliceOfInterfaces(ins)
	if err != nil {
		return nil, merry.Wrap(ErrCannotProcessDepositTxInput, merry.WithCause(err))
	}

	result := make([]interface{}, 0, len(insSlice))

	for _, in := range insSlice {
		res, err := ToResult(in)
		if err != nil {
			result = append(result, Result{
				Err:   errors.Wrap(ErrCannotConvertInterfaceToResult),
				Value: nil,
			})

			continue
		}

		foo, err := foo(res.Value)
		if err != nil {
			result = append(result, Result{
				Err:   errors.Wrap(ErrCannotRunFoo),
				Value: nil,
			})

			continue
		}

		bar, err := bar(ctx, foo)
		if err != nil {
			result = append(result, Result{
				Err:   errors.Wrap(ErrCannotRunBar),
				Value: nil,
			})

			continue
		}

		result = append(result, Result{Err: nil, Value: bar})
	}

	return result, nil
}
```

Any help or advice is appreciated.

**Comment by marksalpeter** [comment]
Is this a similar issue to the one in #4?

**Comment by marksalpeter** [comment]
Closed in favor of #4
